src/Infiltration/formulas/game.ts

```diff
@@ -5,35 +5,41 @@ export const MaxDifficultyForInfiltration = 3.5;
 // This value is typically denoted "lambda," and is the instantaneous rate of decay.
 // I.e. infils decay towards 0 at 0.1%/sec. This is a half-life of ~11.5 minutes.
 const DecayRate = -1e-6;
+// Tuning factor for the market demand multiplier.
+// Math.sqrt(1/TuningMult) infiltrations in a short window will reduce the rewards to near zero for a time.
+const TuningMult = 1.5e-2;
+const date = globalThis.Date;
 
 export const InfiltrationStateDefault = {
   lastChangeTimestamp: 0,
   infils: 0,
 };
-// Tracks an exponential moving average of number of successful infiltrations performed,
-// which decays back to 0. This state is only updated after a successful infil.
+// Tracks an exponential moving average of number of successful infiltrations performed, which decays back to 0.
 export const InfiltrationState = { ...InfiltrationStateDefault };
 
-function calculateCurrentInfils(timestamp: number): number {
-  return InfiltrationState.infils * Math.exp(DecayRate * (timestamp - InfiltrationState.lastChangeTimestamp));
+function calculateCurrentInfils(timestamp = date.now()): number {
+  // The clamp ensures that the market rate is always recovering, and not "stuck" at zero for a long time due to being effectively negative.
+  const clampedInfils = clampNumber(InfiltrationState.infils, 0.01, Math.sqrt(1 / TuningMult) + 0.07);
```

Comment:
This is the wrong place for this action to happen. Infils is a moving average that tracks the number of infiltrations over time; in particular, clamping it to 0.01 so that it can never get down to 0 is very bad.

Leave the logic in here alone, and change the calculation of market demand multiplier (putting in clamps on infil or whatever) if you want to do this.